### PR TITLE
Rename `fold` to reflect actual purpose

### PR DESCRIPTION
**Pull request by AlexGrafl**

### CODE REVIEW

File: util-network/src/main/kotlin/com/tailoredapps/androidutil/network/emptynetworkresponse/EmptyNetworkResponseExt.kt [hunk 1]
```diff
@@ -53,6 +53,7 @@ fun Observable<EmptyNetworkResponse>.split(): Observable<Unit> {
  * Extension function that folds a [EmptyNetworkResponse] to a wrapped type that has an success and an error state.
  */
 @CheckReturnValue
+@Deprecated("Renamed function to reflect it's actual functionality", replaceWith = ReplaceWith("this.wrap(successCreator, errorCreator)"))
```
Comment (FlowMo7) [inline util-network/src/main/kotlin/com/tailoredapps/androidutil/network/emptynetworkresponse/EmptyNetworkResponseExt.kt:56]: Nice usage of the `@Deprecated` annotation with editor actions :+1:

**Review by FlowMo7 — APPROVED** [review]
:+1: 

Not really part of this PR, but I would propose to add a changelog somewhere which makes the changes from version to version more transparent.
I would either propose a markdown-file in the root of this repo (something like `CHANGELOG.md`), which contains the versions with the respective changes, or to create Github-Releases on each version-tag, and add the respective changes there.

**Review by floschu — APPROVED** [review]
LGTM

File: util-network/src/main/kotlin/com/tailoredapps/androidutil/network/emptynetworkresponse/EmptyNetworkResponseExt.kt (hunk 1 above)
Comment (AlexGrafl) [reply util-network/src/main/kotlin/com/tailoredapps/androidutil/network/emptynetworkresponse/EmptyNetworkResponseExt.kt:56]: 👌

**Pull request merged**